packages/svelte/src/internal/client/dom/legacy/lifecycle.js

```diff
@@ -1,21 +1,46 @@
 /** @import { ComponentContextLegacy } from '#client' */
 import { run, run_all } from '../../../shared/utils.js';
+import { derived } from '../../reactivity/deriveds.js';
 import { user_pre_effect, user_effect } from '../../reactivity/effects.js';
 import { component_context, deep_read_state, get, untrack } from '../../runtime.js';
 
 /**
  * Legacy-mode only: Call `onMount` callbacks and set up `beforeUpdate`/`afterUpdate` effects
+ * @param {boolean} [immutable]
  */
-export function init() {
+export function init(immutable = false) {
 	const context = /** @type {ComponentContextLegacy} */ (component_context);
 
 	const callbacks = context.l.u;
 	if (!callbacks) return;
 
+	let props = () => deep_read_state(context.s);
+
+	if (immutable) {
+		let version = 0;
```

Comment:
Shouldn't this be a source?

Reply:
No, it's used in the derived, and the derived then does equality checks to see whether or not the version has changed. If we had the possibility of accessing the current/previous value within the derived I would do that instead.